core/io/json.cpp

```diff
@@ -106,7 +107,10 @@ String JSON::_print_var(const Variant &p_var, const String &p_indent, int p_cur_
 					s += ",";
 					s += end_statement;
 				}
-				s += _make_indent(p_indent, p_cur_indent + 1) + _print_var(String(E->get()), p_indent, p_cur_indent + 1, p_sort_keys);
+				String key_str;
+				VariantWriter::write_to_string(E->get(), key_str);
```

Comment:
Because we need to parse it later with `VariantParser`. String(Variant) doesn't seem to be intended for serialization purposes. This wraps the string in `\n`.